### PR TITLE
PLAT-1786 updated ucm-library and added support for both Filter and C…

### PR DESCRIPTION
**Pull request by CristianDina**
#### Definition of Done:
- [ ] Is there appropriate test coverage?
- [ ] Does the confluence need to be updated?
- [ ] Does this add new dependencies? If so, have the healthchecks been updated?
- [ ] Any Deployment changes we need to be aware of? New infrastructure? Changes to existing components?
- [ ] Is there appropriate logging included?
- [ ] Does it imply any kind of data migration? If so what are the steps for that?
- [ ] Is the Documentation been updated, if needed?

### CODE REVIEW

File: src/main/scala/uk/co/telegraph/ucm/storage/routes/UcmStorageRoute.scala [hunk 1]
```diff
@@ -49,7 +49,19 @@ class UcmStorageRoute @Inject()
...
+    def isEmpty(filter: Option[UCMFilter]): Boolean = {
+      val emptySet: Set[UCMCriteria] = Set.empty
+      filter match {
+        case Some(UCMFilter(Some(emptySet),None,None,None)) => true
+        case _ => false
+      }
+    }
+
+    val criteria: Option[Set[UCMCriteria]] = if (isEmpty(filter))
+      req.body.asJson
```
Comment (athermughal) [inline src/main/scala/uk/co/telegraph/ucm/storage/routes/UcmStorageRoute.scala]: Add a unit test where ucmfilter is empty and you test creation of ucm criteria filter

File: src/test/scala/uk/co/telegraph/ucm/storage/routes/UcmStorageRouteSpec.scala [hunk 2]
```diff
@@ -402,16 +402,16 @@ class UcmStorageRouteSpec
...
-        val filter: UCMFilter = buildFilter(emptyCriteria, None)
-        val expectedScanRequest: ScanCriteriaRequest = buildScanRequestWithActiveFlagFiltering(filter)
-        storageClient.scanUcm _ when expectedScanRequest returns successful(ScanResult(0, Seq.empty))
-
-        val result = endpoint.scan(1, 0)(request.withJsonBody(filter))
-
-        status(result) shouldBe OK
-        toObject[ListResponse](result) shouldBe sampleListResponseWithoutItems
-      }
+//      "should be able to ignore the criteria if no terms or no nested queries are supplied and active flag is missing" in {
```
Comment (athermughal) [inline src/test/scala/uk/co/telegraph/ucm/storage/routes/UcmStorageRouteSpec.scala]: why this test is disabled?

Comment (CristianDina) [reply src/test/scala/uk/co/telegraph/ucm/storage/routes/UcmStorageRouteSpec.scala]: Re-enabled

File: src/main/scala/uk/co/telegraph/ucm/storage/routes/UcmStorageRoute.scala (hunk 1 above)
Comment (CristianDina) [reply src/main/scala/uk/co/telegraph/ucm/storage/routes/UcmStorageRoute.scala]: Added tests
